statsd/reporter.go

```diff
@@ -64,6 +85,60 @@ func (r *cactusStatsReporter) ReportTimer(name string, tags map[string]string, i
 	r.statter.TimingDuration(name, interval, r.sampleRate)
 }
 
+func (r *cactusStatsReporter) ReportHistogramValueSamples(
+	name string,
+	tags map[string]string,
+	buckets tally.Buckets,
+	bucketLowerBound,
+	bucketUpperBound float64,
+	samples int64,
+) {
+	r.statter.Inc(
+		fmt.Sprintf("%s.%s-%s", name,
+			r.valueBucketString(bucketLowerBound),
+			r.valueBucketString(bucketUpperBound)),
+		samples, r.sampleRate)
+}
+
+func (r *cactusStatsReporter) ReportHistogramDurationSamples(
+	name string,
+	tags map[string]string,
+	buckets tally.Buckets,
+	bucketLowerBound,
+	bucketUpperBound time.Duration,
+	samples int64,
+) {
+	r.statter.Inc(
+		fmt.Sprintf("%s.%s-%s", name,
+			r.durationBucketString(bucketLowerBound),
+			r.durationBucketString(bucketUpperBound)),
+		samples, r.sampleRate)
+}
+
+func (r *cactusStatsReporter) valueBucketString(
+	upperBound float64,
+) string {
+	if upperBound == math.MaxFloat64 {
+		return "infinity"
```

Comment:
For code readability I'm inclined to leave them here.  They will be consts for essentially after compilation.